runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonScript.java

```diff
@@ -0,0 +1,501 @@
+package net.runelite.client.plugins.microbot.zerozero.tormenteddemons;
+
+import lombok.SneakyThrows;
+import net.runelite.api.HeadIcon;
+import net.runelite.api.NPC;
+import net.runelite.api.Skill;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.plugins.microbot.Microbot;
+import net.runelite.client.plugins.microbot.Script;
+import net.runelite.client.plugins.microbot.util.bank.Rs2Bank;
+import net.runelite.client.plugins.microbot.util.equipment.Rs2Equipment;
+import net.runelite.client.plugins.microbot.util.gameobject.Rs2GameObject;
+import net.runelite.client.plugins.microbot.util.Rs2InventorySetup;
+import net.runelite.client.plugins.microbot.util.grounditem.LootingParameters;
+import net.runelite.client.plugins.microbot.util.grounditem.Rs2GroundItem;
+import net.runelite.client.plugins.microbot.util.inventory.Rs2Inventory;
+import net.runelite.client.plugins.microbot.util.npc.Rs2Npc;
+import net.runelite.client.plugins.microbot.util.player.Rs2Player;
+import net.runelite.client.plugins.microbot.util.prayer.Rs2Prayer;
+import net.runelite.client.plugins.microbot.util.prayer.Rs2PrayerEnum;
+import net.runelite.client.plugins.microbot.util.walker.Rs2Walker;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class TormentedDemonScript extends Script {
+
+    public static final double VERSION = 1.0;
+    private boolean isRunning = false;
+    public static int killCount = 0;
+    private Rs2PrayerEnum currentDefensivePrayer = null;
+    private Rs2PrayerEnum currentOffensivePrayer = null;
+    private HeadIcon currentOverheadIcon = null;
+    private NPC currentTarget;
+    private boolean lootAttempted = false;
+    private String lastChatMessage = "";
+    private boolean isRestocking = false;
+
+
+    private static final int MAGIC_ATTACK_ANIMATION = 11388;
+    private static final int RANGE_ATTACK_ANIMATION = 11389;
+    private static final int MELEE_ATTACK_ANIMATION = 11392;
+
+    private static final WorldPoint SAFE_LOCATION = new WorldPoint(3150, 3634, 0);
+
+    private enum State { BANKING, TRAVEL_TO_TORMENTED, FIGHTING }
+    public static State BOT_STATUS = State.BANKING;
+
+    private enum TravelStep { LOCATION_ONE, CLIMB_FIRST_STAIRS, CLIMB_SECOND_STAIRS, CLIMB_THROUGH, LOCATION_THREE }
+    private TravelStep travelStep = TravelStep.LOCATION_ONE;
+
+    private enum BankingStep { DRINK, BANK, LOAD_INVENTORY }
+    private BankingStep bankingStep = BankingStep.DRINK;
+
+    public boolean run(TormentedDemonConfig config) {
+        if (config.fullAuto()) {
```

Comment:
Since only one of the two options is possible I think it's more logical to have an enum in your config with the available options. This will then display as a dropdown.

Example in tithe farm script: 

![image](https://github.com/user-attachments/assets/1da6f623-fb42-4592-ba3d-61006ad581e3)

Reply:
All done